### DISCUSSION TITLE
v1.8.3 -> v2 TS + Http API issue (moving over from closed issue thread)

### DISCUSSION THREAD

**Discussion by AdamAH** [serverless/serverless#10532] [Q&A] [answered]
> Hello @AdamAH - make sure you're using the latest version of the Framework - what version are you on? Could you please post a reproducible example and output of `sls deploy` as well?

Gotcha - i was on 1.83.3 - this was from the boilerplate apps created when i setup project a month ago (ts + http API). I updated it to `frameworkVersion: "2"` and everything broke. All my routes just return `null` now so I had to revert back.

I tried installing serverless offline to test but nothing everything is fine here. Not sure how to debug this without impacting the environment

_Originally posted by @AdamAH in https://github.com/serverless/serverless/issues/8210#issuecomment-1014862907_

**Comment by AdamAH** [comment]
@pgrzesik moving to new issue

**Comment by pgrzesik** [comment]
I'm going to move this to discussions as it doesn't seem to be a bug report

**Comment by pgrzesik** [comment]
Hey @AdamAH at this point it's really hard to suggest anything. Are you able to provide reproducible example plus detailed output of failing `sls deploy` command?

**Reply by AdamAH** [reply]
hey @pgrzesik - output of deploy is the exact same for both versions. I can repro it consistently by removing the local version of serverless from node_modules (which is 1.83.3) so it uses global version (2.69.1). 

Deploy succeeds, everything looks fine, only 2 things that drastically change are:

1. every httpAPI connected to a lambda route returns null on every route
2. the code size goes from 1.4MB to 20MB with v2


Based on the docs, I dont believe I'm using anything specific to v1 for setup. Sharing a high level of my template below:

> `frameworkVersion: "1.83.3"
> 
> provider:
>   name: aws
>   runtime: nodejs14.x
>   lambdaHashingVersion: "20201221"
>   vpc:
>     securityGroupIds:
>       - sg-XXXXXXXXXXXXXX
>     subnetIds:
>       - subnet-XXXXXXXXXXXXXX
>       - subnet-XXXXXXXXXXXXXX
> 
> functions:
>   data:
>     handler: handlers/data/router.route
>     events:
>       - httpApi:
>           path: /data/rates
>           method: get
>       - httpApi:
>           path: /data/sample
>           method: get
> 
> plugins:
>   - serverless-plugin-typescript
>   - serverless-offline
> `

I am using typescript so that gets transpiled. 

The project was started from the apps page on serverless.com by selecting httpAPI with typescript. That was the version in the app, and this was done back on Dec 15 2021, so just over a month ago, not a long time ago. Are there constraints with using TS with v2?

Thank you again for your help

**Reply by pgrzesik** [reply]
Thanks for sharing that @AdamAH - in that case the best bet would be to take a look at the breaking changes introduced in `v2` and see if something might've been impacted in your case. 

There are no constraints with `v2` - in fact, `v1` is currently considered obsolete.

**Comment by AdamAH** [comment, marked as answer]
Thanks for that @pgrzesik  - I checked release notes and looks like there was a change from HTTP API responses moving from 1.0 to 2.0 and params changed there. So I'm addressing those changes. Good to go on my end now!

**Reply by pgrzesik** [reply]
Great to hear that 👍
